torch/csrc/jit/import.cpp

```diff
@@ -303,6 +309,13 @@ void ScriptModuleDeserializer::convertModule(
         tensor_table_,
         import_callback);
   }
+
+  // TODO: get the correct index so this works for submodules
+  static size_t module_num = 0;
+  if (module_num < state_table_.size()) {
+    module->setstate(state_table_.at(module_num));
+  }
+  module_num++;
```

Comment:
Similar to the reasons for attributes containing an ID, this should not simply update a number. It is hard to follow and couples the serialization format with the deserialization order. What if we wanted to only deserialize one submodule?